### PR TITLE
Medium complexity assignment 1 - Employee Tax

### PR DESCRIPTION
**Pull request by AmrutaDhage20**

### CODE REVIEW

File: src/com/encora/javaAssignments/MediumComplexityProblems/employeeTax/EmployeeTax.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    /* Java program to calculate income tax */
+
+    public static void main(String args[]) {
+
+        double income = 2000000, tax = 0;
+        double taxableAmount = 0;
+        double taxFor10Percent = 0, taxFor15Percent = 0, taxFor20Percent = 0, taxFor30Percent = 0;
+        double totalTax;
+
+        if (income <= 700000) {
```
Comment (amitsahu8) [inline src/com/encora/javaAssignments/MediumComplexityProblems/employeeTax/EmployeeTax.java]: use switch case

**Review by amitsahu8 — CHANGES_REQUESTED** [review]

File: src/com/encora/javaAssignments/MediumComplexityProblems/employeeTax/EmployeeTax.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+            }
+            if (income > 1500000) {
+                taxFor15Percent = 0.15 * 300000;
+                taxFor20Percent = 0.20 * 500000;
+                taxFor30Percent = (0.30 * (taxableAmount - 300000 - 200000 - 500000));
+                System.out.println("taxFor15Percent: " + taxFor15Percent);
+                System.out.println("taxFor20Percent: " + taxFor20Percent);
+                System.out.println("taxFor30Percent: " + taxFor30Percent);
+            }
+        }
```
Comment (prashantkawa) [inline src/com/encora/javaAssignments/MediumComplexityProblems/employeeTax/EmployeeTax.java]: Try writing logic in method and call that method from main method

**Review by prashantkawa — CHANGES_REQUESTED** [review]

File: src/com/encora/javaAssignments/MediumComplexityProblems/employeeTax/EmployeeTaxUsingSwitchCase.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+public class EmployeeTaxUsingSwitchCase {
+
+    public static void calculateIncomeTax(double salary) {
+
+        double taxableAmount = 0;
+        int taxSlab;
+        double taxFor10Percent = 0, taxFor15Percent = 0, taxFor20Percent = 0, taxFor30Percent = 0;
+
+        taxSlab = (salary > 700000 && salary <= 1000000) ? 2 : (salary > 1000000 && salary <= 1500000) ? 3 : (salary > 1500000) ? 4 : 1;
```
Comment (prashantkawa) [inline src/com/encora/javaAssignments/MediumComplexityProblems/employeeTax/EmployeeTaxUsingSwitchCase.java]: please verify the conditions again and change it accordingly

File: src/com/encora/javaAssignments/MediumComplexityProblems/employeeTax/EmployeeTaxUsingSwitchCase.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+            }
+            case 4: {
+                taxFor10Percent = 0.1 * 200000;
+                taxFor15Percent = 0.15 * 300000;
+                taxFor20Percent = 0.20 * 500000;
+                taxFor30Percent = (0.30 * (taxableAmount - 300000 - 200000 - 500000));
+                System.out.println("taxFor10Percent: " + taxFor10Percent);
+                System.out.println("taxFor15Percent: " + taxFor15Percent);
+                System.out.println("taxFor20Percent: " + taxFor20Percent);
+                System.out.println("taxFor30Percent: " + taxFor30Percent);
```
Comment (prashantkawa) [inline src/com/encora/javaAssignments/MediumComplexityProblems/employeeTax/EmployeeTaxUsingSwitchCase.java]: just print total tax amount

**Review by prashantkawa — CHANGES_REQUESTED** [review]

File: src/com/encora/javaAssignments/MediumComplexityProblems/employeeTax/EmployeeTax.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+    public static void calculateIncomeTax(double salary) {
+
+        double taxableAmount = 0;
+        int taxSlab = 0;
+        double tax = 0;
+
+        taxSlab = (salary > 700000 && salary <= 1000000) ? 2 : (salary > 1000000 && salary <= 1500000) ? 3 : (salary > 1500000) ? 4 : 1;
+        System.out.println("taxSlab: " + taxSlab);
+
```
Comment (rohitlamsonge66) [inline src/com/encora/javaAssignments/MediumComplexityProblems/employeeTax/EmployeeTax.java]: remove extra spaces

File: src/com/encora/javaAssignments/MediumComplexityProblems/employeeTax/employeeTax.README.md [hunk 6]
```diff
@@ -0,0 +1,9 @@
+Q.1 Write a java program to find tax for employee. The program should  
+calculate the income tax to be paid by the employee as per the criteria given below: 
+Slab rate                   IT rate 
+Upto Rs. 7,00,000           Nil 
+Above Rs. 7,00,000          (0-5,00,000=nil,5,00,000 – 7,00,000=10%) 
+Rs. 7,00,000-10,00,000      15% on additional amount 
+Rs. 10,00,000-15,00,000     20% on additional amount 
+Above 15,00,000           30% on additional amount 
+Result: - income tax is …………………………….
```
Comment (rohitlamsonge66) [inline src/com/encora/javaAssignments/MediumComplexityProblems/employeeTax/employeeTax.README.md]: add output in read me.

**Review by prashantkawa — APPROVED** [review]
Approving PR

File: src/com/encora/javaAssignments/MediumComplexityProblems/employeeTax/EmployeeTax.java [hunk 7]
```diff
@@ -0,0 +1,30 @@
+package MediumComplexityProblems.employeeTax;
+public class EmployeeTax {
+    /* Java program to calculate Income Tax */
+    public static void calculateIncomeTax(double salary) {
+        double taxableAmount = 0;
+        int taxSlab = 0;
+        double tax = 0;
+        taxSlab = (salary <= 700000) ? 1 : (salary > 700000 && salary <= 1000000) ? 2 : (salary > 1000000 && salary <= 1500000) ? 3 : 4;        System.out.println("taxSlab: " + taxSlab);
```
Comment (amitsahu8) [inline src/com/encora/javaAssignments/MediumComplexityProblems/employeeTax/EmployeeTax.java]: sout should be in new line

File: src/com/encora/javaAssignments/MediumComplexityProblems/employeeTax/EmployeeTax.java [hunk 8]
```diff
@@ -0,0 +1,30 @@
+package MediumComplexityProblems.employeeTax;
+public class EmployeeTax {
```
Comment (amitsahu8) [inline src/com/encora/javaAssignments/MediumComplexityProblems/employeeTax/EmployeeTax.java:3]: space missing between package and method

**Review by amitsahu8 — CHANGES_REQUESTED** [review]

**Review by amitsahu8 — APPROVED** [review]

**Pull request merged**